api/server/router/build/build_routes.go

```diff
@@ -67,6 +69,10 @@ func newImageBuildOptions(ctx context.Context, r *http.Request) (*types.ImageBui
 		options.Isolation = i
 	}
 
+	if runtime.GOOS != "windows" && options.SecurityOpt != nil {
+		return nil, fmt.Errorf("the daemon on this platform does not support --security-opt to build")
```

Comment:
For the status code, just https://github.com/docker/docker/blob/master/api/errors/errors.go#L21